rpc/codec.go

```diff
@@ -26,13 +27,15 @@ import (
 type NodeAwareServerCodec struct {
 	rpc.ServerCodec
 	NodeID *proto.RawNodeID
+	Ctx    context.Context
 }
 
 // NewNodeAwareServerCodec returns new NodeAwareServerCodec with normal rpc.ServerCode and proto.RawNodeID
-func NewNodeAwareServerCodec(codec rpc.ServerCodec, nodeID *proto.RawNodeID) *NodeAwareServerCodec {
+func NewNodeAwareServerCodec(codec rpc.ServerCodec, nodeID *proto.RawNodeID, ctx context.Context) *NodeAwareServerCodec {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
context.Context should be the first parameter of a function